### PR TITLE
Convert JWT-Knox into a ViewSet

### PR DESCRIPTION
**Pull request by ssaavedra**
This converts the different APIViews that formed the JWT-Knox app into
a single ViewSet.

The advantages of this approach are: less code repetition (although we
managed to avoid some by subclassing the APIView in the first place),
and the use of a router, which may aid the SchemaGenerator when we
generate an API self-description to know that several endpoints are
related to the same API behaviour.

### CODE REVIEW

**Comment by ssaavedra** [comment]
WARNING: In the current state, this does not pass the defined tests. Those need to get updated to work with the ViewSet.

**Comment by ssaavedra** [comment]
Now the tests should pass, but I want to squash the commits before doing the merge. The commits are now spread so that they are easier to inspect. If it looks good to you, I'll squash them for the merge.

cc/ @castrinho8 ¿a quien más hago ping?

File: xea-core/jwt_knox/viewsets.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        """
+        token = create_auth_token(user=request.user, expires=expires)
+        return Response(response_payload_handler(token, request.user, request))
+
+    @list_route(methods=('get', 'post'))
+    def verify(self, request):
+        """
+        This view allows a third party to verify a web token.
+        """
+        return Response(None, status=status.HTTP_204_NO_CONTENT)
```
Comment (castrinho8) [inline xea-core/jwt_knox/viewsets.py:78]: Esto está implementado?

File: xea-core/jwt_knox/viewsets.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    def get_authenticators_for_view(self, view_name):
+        """
+        Define the authenticators present in a specific view.
+        """
+        pass
+
+
+class JWTKnoxAPIViewSet(PerViewAuthenticatorMixin, ViewSet):
+    """This API endpoint set enables authentication via **JSON Web Tokens** (JWT).
+
```
Comment (castrinho8) [inline xea-core/jwt_knox/viewsets.py:46]: Chorrada de identación.

Por cierto, en los @list_route acaban en ",", es normal?

**Comment by therealtod** [comment]
lgtm

**Comment by ssaavedra** [comment]
LGTM

**Pull request merged**